### PR TITLE
Add Github Actions

### PR DESCRIPTION
**Pull request by HotaruBlaze**
<ins>**This needs #68 or the macOS build will be broken!**</ins>

---
This PR adds auto building for NSO-RPC via GitHub Actions, I've tested all 3 builds and they seem to work correctly, however NSO-RPC's mac build still needs tested on an m1 mac, but I believe it should work without an issue due to Rosetta 2.

This does not currently set the release notes, however it does not set these and only adds the files as their build, so that is currently needed to be done manually.

This also changes how NSO-RPC is build on mac slightly. This changes the original `mac.zip` to `mac-portable.zip`, as I've added an installable version of NSO-RPC via productbuild and that's known as `mac-installer.zip`, More details about my reasoning below.

---

This is mainly useful because for #66, this PR isn't needed for it to work, however if NSO-RPC is set to autorun from the downloads folder, it will usually not work due to being in a security folder instead of a normal one. 

However if moved to another folder, such as desktop it works without issues.

TLDR:
- mac-portable autostart works if <ins>**not running**</ins> from the downloads folder.
- mac-installer will install NSO-RPC in /Applications and thus autostart works correctly.

Note: the installer is just a packaged version of the portable that's simply installed into /Applications

---

Extra Note: I could have probably cleaned up the Github action little more, however I wanted to try and keep it simple and at most would have just removed the skip's when looking at the individual builds, however that's just visual.

### CODE REVIEW

**Comment by MCMi460** [comment]
The time that it takes to make a release has now dwindled to an infinitesimally small period.
Thank you so so much for doing this!
